modules/apps/forms-and-workflow/calendar/calendar-service/src/main/java/com/liferay/calendar/internal/upgrade/v1_0_6/UpgradeResourcePermissions.java

```diff
@@ -0,0 +1,194 @@
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.security.permission.ResourceActionsImpl;
+import com.liferay.portal.util.PropsValues;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import java.util.List;
+
+/**
+ * @author José María Muñoz
+ * @author Alberto Chaparro
+ */
+public class UpgradeResourcePermissions extends UpgradeProcess {
+
+	public UpgradeResourcePermissions(
+		ResourceActionLocalService resourceActionLocalService,
+		ResourceBlockLocalService resourceBlockLocalService,
+		RoleLocalService roleLocalService) {
+
+		_resourceActionLocalService = resourceActionLocalService;
+		_resourceBlockLocalService = resourceBlockLocalService;
+		_roleLocalService = roleLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		upgradeGuestResourcePermissions();
+	}
+
+	protected long getCalendarResourceUnsupportedActionsBitwiseValue()
+		throws PortalException {
+
+		int unsupportedBitwiseValue = 0;
+
+		List<String> guestUnsupportedActions =
+			getModelResourceGuestUnsupportedActions();
+
+		for (String resourceActionId : _NEW_UNSUPPORTED_ACTION_IDS) {
+			if (guestUnsupportedActions.contains(resourceActionId)) {
+				ResourceAction resourceAction =
+					_resourceActionLocalService.getResourceAction(
+						_CALENDAR_RESOURCE_NAME, resourceActionId);
+
+				unsupportedBitwiseValue |= resourceAction.getBitwiseValue();
+			}
+		}
+
+		return unsupportedBitwiseValue;
+	}
+
+	protected List<String> getModelResourceGuestUnsupportedActions()
+		throws UpgradeException {
+
+		try {
+			ResourceActionsImpl resourceActionsImpl = new ResourceActionsImpl();
+
+			Class<?> clazz = getClass();
+
+			ClassLoader classLoader = clazz.getClassLoader();
+
+			for (String config : PropsValues.RESOURCE_ACTIONS_CONFIGS) {
+				resourceActionsImpl.read(null, classLoader, config);
+			}
+
+			return resourceActionsImpl.getModelResourceGuestUnsupportedActions(
+				_CALENDAR_RESOURCE_NAME);
+		}
+		catch (Exception e) {
+			throw new UpgradeException(e);
+		}
+	}
+
+	protected void upgradeGuestResourcePermissions() throws Exception {
+		long unsupportedBitwiseValue =
+			getCalendarResourceUnsupportedActionsBitwiseValue();
+
+		if (unsupportedBitwiseValue == 0) {
+			return;
+		}
+
+		StringBundler sb = new StringBundler(2);
+
+		sb.append("select companyId, groupId, calendarResourceId, ");
+		sb.append("resourceBlockId from CalendarResource");
+
+		try (PreparedStatement ps =
+				connection.prepareStatement(sb.toString())) {
+
+			try (ResultSet rs = ps.executeQuery()) {
+				while (rs.next()) {
+					long companyId = rs.getLong(1);
+					long groupId = rs.getLong(2);
+					final long primaryKey = rs.getLong(3);
+					final long resourceBlockId = rs.getLong(4);
+
+					Role guestRole = _roleLocalService.getRole(
+						companyId, RoleConstants.GUEST);
+
+					PermissionedModel permissionedModel =
```

Comment:
Hey @brandizzi!,

Thanks for your work and effort reviewing this but your changes are not working and I'll tell you why:

If you use your gogo console to reviewing dependencies (scr:list and scr:info commands), you will see that CalendarResourceLocalService is 'Unsatisfied':
```
UnsatisfiedReference: CalendarResourceLocalService
      Target: null
      (no target services)
```
This mean that we can not use `CalendarResourceLocalService` and the reason is because our code is into `CalendarResourceLocalService`, so, this service will never be initialized.

That's why we need to use `PermissionedModel`, because we need to invoke methods with CalendarResource `PermissionedModel` but we can not use `CalendarResourceLocalService`.

In addition, speaking with @achaparro (SME for Upgrades), told me that we should use SQL Statements for tables of the service that we are upgrading (because it suppose that this service know how to upgrade his tables) and use local services API for the rest of tables updates (because we let other services that update their own tables).

Please let me know if you need more information.

Thanks!